src/main/java/io/quarkus/gizmo2/impl/Neg.java

```diff
@@ -26,12 +26,12 @@ public boolean bound() {
     }
 
     public void writeCode(final CodeBuilder cb, final BlockCreatorImpl block) {
-        switch (typeKind()) {
+        switch (typeKind().asLoadable()) {
             case INT -> cb.ineg();
             case LONG -> cb.lneg();
             case FLOAT -> cb.fneg();
             case DOUBLE -> cb.dneg();
-            default -> throw new IllegalStateException();
+            default -> throw new IllegalStateException("Cannot negate non-primitive expression: " + a);
```

Comment:
For validations, we should do them up front (in `BlockCreatorImpl#neg()` in this case). Otherwise what happens is that the exception doesn't come up until much later and we lose the line number of the problem.

Reply:
I didn't check if this is validate before, I just stumbled upon an ISE without a message and figured I'll write something at least. I'll check how that's done. If there's an upfront validation, I guess we could throw `AssertionError` here or something like that.

Reply:
ISE is commonly used for invalid switch states IIRC. There's also an `impossibleSwitchCase()` in `smallrye-common-constraint` that might be a good choice here.

Reply:
That makes sense.

Reply:
Done. Added argument validation to the constructor, similarly to `Rel`.